src/tribler-core/tribler_core/components/tests/test_tribler_components.py

```diff
@@ -64,6 +63,25 @@ async def test_ipv8_component(tribler_config):
 
         await session.shutdown()
 
+    # test dht disabled
+    tribler_config.dht.enabled = True
+    session = Session(tribler_config, [MasterKeyComponent(), RESTComponent(), Ipv8Component()])
+    with session:
+        await session.start()
+
+        comp = Ipv8Component.instance()
+        assert comp.dht_discovery_community
+
+    # test discovery_community enabled
+    tribler_config.gui_test_mode = False
+    tribler_config.discovery_community.enabled = True
+    session = Session(tribler_config, [MasterKeyComponent(), RESTComponent(), Ipv8Component()])
+    with session:
+        await session.start()
+
+        comp = Ipv8Component.instance()
+        assert comp._peer_discovery_community
```

Comment:
**suggestion:** This test now breaks with the `Arrange/Act/Assert` pattern: what I see here is `Arrange/Assert/Arrange/Assert/Arrange/Assert`. By splitting this code at the comments (turning `# test dht disabled` into `async def test_ipv8_component_dht_disabled()` and `# test discovery_community enabled` into `async def test_ipv8_component_discovery_enabled()`) you can follow the AAA pattern with very little effort.

Reply:
Agree! Will implement it soon.

Reply:
Implemented